### PR TITLE
svelte-language-server: Use buildNpmPackage

### PR DESCRIPTION
**Pull request by purepani**
## Description of changes
See https://github.com/NixOS/nixpkgs/issues/229475
Vendors a generated lock file since upstream uses pnpm. Since upstream uses a workspace as well, the lock file is retrieved from the npm registry instead of directly from source due to peer dependencies. 
Also updates from 0.16.5 to 0.16.9.
The entry from pkgs/development/node-packages/node-packages.json was removed, but packages were breaking  with errors similar to https://github.com/NixOS/nixpkgs/pull/305026, so node-packages was not bumped. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Not sure who to ping since there are no code owners, but it looks like @dotlambda and @SuperSandro2000 have been merging a lot of these npm packaging changes; let me know if there's someone else I should notify.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by purepani** [comment]
Result of `nixpkgs-review pr 311260` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>svelte-language-server</li>
  </ul>
</details>

File: pkgs/by-name/sv/svelte-language-server/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  dontNpmBuild = true;
+
+  passthru.updateScript = ./update.sh;
+
+  meta = {
+    description = "A language server (implementing the language server protocol) for Svelte.";
+    downloadPage = "https://www.npmjs.com/package/svelte-language-server";
+    homepage = "https://github.com/sveltejs/language-tools";
+    license = lib.licenses.mit; # no license specified
```
Comment (SuperSandro2000) [inline pkgs/by-name/sv/svelte-language-server/package.nix:30]:
```suggestion
    license = lib.licenses.mit;
```
there is a license https://github.com/sveltejs/language-tools/blob/master/package.json#L5

File: pkgs/by-name/sv/svelte-language-server/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  postPatch = ''
+    ln -s ${./package-lock.json} package-lock.json
+  '';
+
+  dontNpmBuild = true;
+
+  passthru.updateScript = ./update.sh;
+
+  meta = {
+    description = "A language server (implementing the language server protocol) for Svelte.";
```
Comment (SuperSandro2000) [inline pkgs/by-name/sv/svelte-language-server/package.nix:27]:
```suggestion
    description = "Language server (implementing the language server protocol) for Svelte";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by purepani** [comment]
Updated with requested changes

**Comment by purepani** [comment]
Hey @SuperSandro2000 is this good to merge?

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**